pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2069,3 +2100,14 @@ type Counter struct {
 	// Value is the value of the metric
 	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
 }
+
+// Memoization
+type Memoize struct {
+	MaxAge string `json:"maxAge" protobuf:"bytes,1,opt,name=maxAge"`
+	Key    string `json:"key" protobuf:"bytes,2,opt,name=key"`
+	Cache  *Cache `json:"cache" protobuf:"bytes,3,opt,name=cache"`
```

Comment:
discussing with @jessesuen about moving to `/spec/memoizationConfig` (or similar)